### DISCUSSION TITLE
Managing System Resources Using Nixpacks

### DISCUSSION THREAD

**Discussion by ghost** [railwayapp/nixpacks#122] [General]
Managing system resources (RAM, stack size, max file watch, CPU time, etc.) using Nixpacks configuration would be a helpful feature. As of now one will have to use custom commands to do that. Making this a core feature will allow users to set limits per phase easily. This can be done by using tools like [`prlimit`](https://manpages.ubuntu.com/manpages/xenial/man1/prlimit.1.html), [`ulimit`](https://ss64.com/bash/ulimit.html) and [`timeout`](https://man7.org/linux/man-pages/man1/timeout.1.html) or by creating our own wrapper. Also, this is how [Piston](https://github.com/engineer-man/piston/blob/master/api/src/job.js#L113-L151) (remote code execution engine) sets resource limits.

**Comment by Milo123459** [comment]
Surely `prlimit` and those tools will require some system dependencies meaning the Dockerfile will get "larger" (thus, taking longer). I think a custom wrapper is a better idea.

**Comment by ghost** [comment]
All of these tools are pre-included in the current base image(`debian:bullseye-slim`).

**Comment by Milo123459** [comment]
Ah, then it is fine to use them.
